Task/ctaa 1374 clean up

dusanjencik opened the pull request:
## Changes

- Removed all code related to P2P.
- Removed all code related to automatic handshakes.
- Removed all code related to manual handshakes.
- Added DB migration to remove all data related to manual and automatic handshake.
- Added preferences migration to remove invalid keys.

## Mergable after
- https://github.com/austrianredcross/stopp-corona-android/pull/85

## Solved issues

- [Issue #1374](https://tasks.pxp-x.com/browse/CTAA-1374)
- [Issue #1480](https://tasks.pxp-x.com/browse/CTAA-1480)

dusanjencik committed:
CTAA-1374 - Removed P2P and Nearby handshakes code

dusanjencik committed:
CTAA-1374 - Reverted automatic start at the first start

dusanjencik committed:
Merge remote-tracking branch 'origin/develop' into task/CTAA-1374_clean_up

TuxCoder commented:
you forgot to remove the maven repo and the api key:

https://github.com/austrianredcross/stopp-corona-android/blob/task/CTAA-1374_clean_up/build.gradle#L26

https://github.com/austrianredcross/stopp-corona-android/blob/task/CTAA-1374_clean_up/app/build.gradle#L63

https://github.com/austrianredcross/stopp-corona-android/blob/task/CTAA-1374_clean_up/app/build.gradle#L92

you can also do `grep p2p -ri <project>` or similar file search tool:

```
./build.gradle
./app/build.gradle
./app/src/main/AndroidManifest.xml
./app/src/main/assets/privacy_en.html
./app/src/main/assets/privacy_de.html
./app/src/main/assets/privacy-and-terms-of-use_en.html
./app/src/main/assets/privacy-and-terms-of-use_de.html

dusanjencik commented:
@TuxCoder yes, thank you very much for spotting it 👍 I'll fix it on Tuesday.

Commit:
CTAA-1547 - permission cleanup

BLUETOOTH is definitly still needed, all other permission should be considered and removed
CTAA-1547

falkorichter commented:
@TuxCoder great comment. Text are beeing changed by the colleagues in the legal department as we speak, permissions will all have to be reconsidered. We have decided to separate these two issues. I moved the Bluetooth permission from the block of p2p permissions and we have a new ticket for reconsideration of all permissions. We might need/use some p2p permissions and want to avoid side affects here...

https://tasks.pxp-x.com/browse/CTAA-1547 is the task :) But you might not have access to it 😢 

🤗  👍  ✋  for your contribution!

Commit:
CTAA-1546 link to discussion added

https://tasks.pxp-x.com/browse/CTAA-1546

Commit:
the service is deleted

Commit:
CTAA-1546 CTAA-1548 TODOs added

https://tasks.pxp-x.com/browse/CTAA-1546
and
https://tasks.pxp-x.com/browse/CTAA-1548

Commit:
CTAA-1374 removal of p2p kit build parameter

relates to https://tasks.pxp-x.com/browse/CTAA-1352

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/constants/Constants.kt:
```diff
@@ -77,7 +77,6 @@ object Constants {
         const val DATA_PRIVACY_REPOSITORY_PREFIX = PREFIX + "terms_and_conditions_repository_"
         const val DASHBOARD_PREFIX = PREFIX + "dashboard_"
         const val QUARANTINE_REPOSITORY_PREFIX = PREFIX + "quarantine_repository_"
-        const val CORONA_DETECTION_REPOSITORY_PREFIX = PREFIX + "corona_detection_repository_"
```
Note to self, are we clearing the value here?

falkorichter replied:
Preference is safe to delete, no data is leaked we can just leave it there... @uluckas or should we be extra safe and delete this old value from system preferences?

dusanjencik replied:
@falkorichter Already removed in https://github.com/austrianredcross/stopp-corona-android/pull/83/files#diff-4d7fe52c6e225b440d9b5efb6841d6a5R33

TuxCoder commented:
@falkorichter  Thanks  for taking care so fast!

Yeah have sadly no access to the internal issue tracker, that makes contribution currently a bit hard....

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt:
```diff
@@ -2,48 +2,26 @@ package at.roteskreuz.stopcorona.model.repositories
-
     /**
-     * Do not show explanation dialog again.
+     * Information if the automatic handshake was enabled automatically on the first start
```
I don´t get this, we don´t have a service anymore...

dusanjencik replied:
Renamed in new commit

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/model/repositories/DashboardRepository.kt:
```diff
@@ -2,48 +2,26 @@ package at.roteskreuz.stopcorona.model.repositories
-    override fun setMicrophoneExplanationDialogShown() {
-        showMicrophoneExplanationDialog = false
-    }
+    override var serviceEnabledOnFirstStart: Boolean by preferences.booleanSharedPreferencesProperty(PREF_SERVICE_ENABLED_ON_FIRST_START, false)
```
this should all be removed IMHO

dusanjencik replied:
not removed, renamed. https://github.com/austrianredcross/stopp-corona-android/pull/83#discussion_r433731286

falkorichter replied:
yes - renamed!

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt:
```diff
@@ -148,7 +147,7 @@ class ExposureNotificationRepositoryImpl(
         }
         registeringWithFrameworkState.loading()
-        exposureNotificationClient.start()
+        exposureNotificationClient.stop()
```
This should have been a fix to the develop branch and not mixed in in here

dusanjencik replied:
Moved to https://github.com/austrianredcross/stopp-corona-android/pull/85

dusanjencik replied:
Updated PR description

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt:
```diff
@@ -189,12 +187,13 @@ class ReportingRepositoryImpl(
-                    nearbyRecord.publicKey to InfectionMessageContent(infectionLevel, nearbyRecord.timestamp)
-                }
-            )
+            // TODO: 28/05/2020 dusanjencik: How to integrate exposure notifications to infection messages?
```
https://wiki.sinnerschrader.com/display/SWP/Best+practices#UNUSED

dusanjencik replied:
If there is todo comment, some code might be commented. Will be fixed in later PR.

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt:
```diff
@@ -202,11 +183,10 @@ class DashboardFragment : BaseFragment(R.layout.fragment_dashboard) {
-//            viewModel.wasServiceEnabledAutomaticallyOnFirstStart = true
-//            checkDependenciesAndStartAutomaticHandshake(true)
-//        }
+        if (viewModel.wasServiceEnabledAutomaticallyOnFirstStart.not()) {
```
wasServiceEnabledAutomaticallyOnFirstStart needs to be renamed. We don´t have no service anymore

Also: I´m unsure if this is a valid operation, unsure if UX approved to register the app automatically for Exposure notifications.

dusanjencik replied:
Renamed https://github.com/austrianredcross/stopp-corona-android/pull/83#discussion_r433731286

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt:
```diff
@@ -231,16 +211,11 @@ class DashboardFragment : BaseFragment(R.layout.fragment_dashboard) {
 
-    // TODO: 27/05/2020 dusanjencik: Remove
-    private fun checkPlayServicesAvailabilityAndStartHandshakeFragment() {
+    private fun checkPlayServicesAvailabilityAndRegisterToExposureNotificationFramework(enableFramework: Boolean) {
```
I believe this should be handled in another Ticket: https://tasks.pxp-x.com/browse/CTAA-1363 and https://tasks.pxp-x.com/browse/CTAA-1395

dusanjencik replied:
Agree. In the code, there is already a todo comment for it: `// TODO: 28/05/2020 dusanjencik: We should check also correct version`

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardFragment.kt:
```diff
@@ -231,16 +211,11 @@ class DashboardFragment : BaseFragment(R.layout.fragment_dashboard) {
-                    startHandshakeFragment()
-                }
+                // TODO: 28/05/2020 dusanjencik: We should check also correct version
+                viewModel.onAutomaticHandshakeEnabled(enableFramework)
```
See above

dusanjencik replied:
Renamed

falkorichter commented on app/src/main/java/at/roteskreuz/stopcorona/screens/dashboard/DashboardViewModel.kt:
```diff
@@ -29,21 +29,11 @@ class DashboardViewModel(
-    fun observeSavedEncounters(): Observable<Int> {
-        return dashboardRepository.observeSavedEncountersNumber()
-    }
+    var wasServiceEnabledAutomaticallyOnFirstStart: Boolean
```
there is no Service anymore

dusanjencik replied:
Done in https://github.com/austrianredcross/stopp-corona-android/pull/83#discussion_r433731286

dusanjencik committed:
CTAA-1374 - Renamed service keyword to exposure framework

dusanjencik committed:
CTAA-1382 - Fixed unregistration functionality

dusanjencik committed:
Merge branch 'task/CTAA-1382_fixed_unregistration' into task/CTAA-1374_clean_up

dusanjencik committed:
CTAA-1374 - Renamed registration method
